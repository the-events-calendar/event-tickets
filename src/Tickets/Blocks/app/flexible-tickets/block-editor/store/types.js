export default {
	SET_DEFAULT_TICKET_DESCRIPTION:
		'TEC-TICKETS-FLEXIBLE-TICKETS/SET_DEFAULT_TICKET_DESCRIPTION',
	SET_MULTIPLE_PROVIDERS_NOTICE_TEMPLATE:
		'TEC-TICKETS-FLEXIBLE-TICKETS/SET_MULTIPLE_PROVIDERS_NOTICE_TEMPLATE',
	SET_SERIES_DATA: 'TEC-TICKETS-FLEXIBLE-TICKETS/SET_SERIES_DATA',
};
