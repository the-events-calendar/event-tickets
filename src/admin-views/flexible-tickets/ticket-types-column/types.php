<?php
/**
 * The template for the ticket types column in the events list table.
 *
 * @since 5.8.0
 *
 * @version 5.8.0
 *
 * @var array $tickets_by_types Array of tickets by types.
 * @var Tribe__Tickets__Admin__Views $admin_views The admin views instance for flexible tickets.
 */
?>
<div class="tec-tickets__series_attached_ticket-types">
	<?php
	foreach ( $tickets_by_types as $type => $tickets ) :
		$admin_views->template( 'ticket-types-column/' . $type, [ 'type'=> $type, 'tickets' => $tickets ] );
	endforeach;
	?>
</div>