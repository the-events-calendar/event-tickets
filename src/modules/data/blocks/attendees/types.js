/* eslint-disable max-len */
/**
 * Internal dependencies
 */
import { PREFIX_TICKETS_STORE } from '@moderntribe/tickets/data/utils';

export const SET_ATTENDEES_INITIAL_STATE = `${ PREFIX_TICKETS_STORE }/SET_ATTENDEES_INITIAL_STATE`;

export const SET_ATTENDEES_TITLE = `${ PREFIX_TICKETS_STORE }/SET_ATTENDEES_TITLE`;

export const SET_ATTENDEES_DISPLAY_TITLE = `${ PREFIX_TICKETS_STORE }/SET_ATTENDEES_DISPLAY_TITLE`;

export const SET_ATTENDEES_DISPLAY_SUBTITLE = `${ PREFIX_TICKETS_STORE }/SET_ATTENDEES_DISPLAY_SUBTITLE`;
