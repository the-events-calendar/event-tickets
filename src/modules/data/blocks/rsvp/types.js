/* eslint-disable max-len */
/**
 * Internal dependencies
 */
import { PREFIX_TICKETS_STORE } from '@moderntribe/tickets/data/utils';

//
// ─── RSVP TYPES ─────────────────────────────────────────────────────────────────
//

export const SET_RSVP_ID = `${ PREFIX_TICKETS_STORE }/SET_RSVP_ID`;
export const SET_RSVP_SETTINGS_OPEN = `${ PREFIX_TICKETS_STORE }/SET_RSVP_SETTINGS_OPEN`;
export const SET_RSVP_HAS_CHANGES = `${ PREFIX_TICKETS_STORE }/SET_RSVP_HAS_CHANGES`;
export const SET_RSVP_IS_LOADING = `${ PREFIX_TICKETS_STORE }/SET_RSVP_IS_LOADING`;
export const SET_RSVP_IS_SETTINGS_LOADING = `${ PREFIX_TICKETS_STORE }/SET_RSVP_IS_SETTINGS_LOADING`;
export const SET_RSVP_IS_MODAL_OPEN = `${ PREFIX_TICKETS_STORE }/SET_RSVP_IS_MODAL_OPEN`;
export const SET_RSVP_GOING_COUNT = `${ PREFIX_TICKETS_STORE }/SET_RSVP_GOING_COUNT`;
export const SET_RSVP_NOT_GOING_COUNT = `${ PREFIX_TICKETS_STORE }/SET_RSVP_NOT_GOING_COUNT`;
export const SET_RSVP_HAS_ATTENDEE_INFO_FIELDS = `${ PREFIX_TICKETS_STORE }/SET_RSVP_HAS_ATTENDEE_INFO_FIELDS`;
export const SET_RSVP_HAS_DURATION_ERROR = `${ PREFIX_TICKETS_STORE }/SET_RSVP_HAS_DURATION_ERROR`;

export const SET_RSVP_DETAILS = `${ PREFIX_TICKETS_STORE }/SET_RSVP_DETAILS`;
export const SET_RSVP_TEMP_DETAILS = `${ PREFIX_TICKETS_STORE }/SET_RSVP_TEMP_DETAILS`;
export const SET_RSVP_HEADER_IMAGE = `${ PREFIX_TICKETS_STORE }/SET_RSVP_HEADER_IMAGE`;

//
// ─── RSVP DETAILS TYPES ─────────────────────────────────────────────────────────
//

export const SET_RSVP_TITLE = `${ PREFIX_TICKETS_STORE }/SET_RSVP_TITLE`;
export const SET_RSVP_DESCRIPTION = `${ PREFIX_TICKETS_STORE }/SET_RSVP_DESCRIPTION`;
export const SET_RSVP_CAPACITY = `${ PREFIX_TICKETS_STORE }/SET_RSVP_CAPACITY`;
export const SET_RSVP_NOT_GOING_RESPONSES = `${ PREFIX_TICKETS_STORE }/SET_RSVP_NOT_GOING_RESPONSES`;
export const SET_RSVP_START_DATE = `${ PREFIX_TICKETS_STORE }/SET_RSVP_START_DATE`;
export const SET_RSVP_START_DATE_INPUT = `${ PREFIX_TICKETS_STORE }/SET_RSVP_START_DATE_INPUT`;
export const SET_RSVP_START_DATE_MOMENT = `${ PREFIX_TICKETS_STORE }/SET_RSVP_START_DATE_MOMENT`;
export const SET_RSVP_END_DATE = `${ PREFIX_TICKETS_STORE }/SET_RSVP_END_DATE`;
export const SET_RSVP_END_DATE_INPUT = `${ PREFIX_TICKETS_STORE }/SET_RSVP_END_DATE_INPUT`;
export const SET_RSVP_END_DATE_MOMENT = `${ PREFIX_TICKETS_STORE }/SET_RSVP_END_DATE_MOMENT`;
export const SET_RSVP_START_TIME = `${ PREFIX_TICKETS_STORE }/SET_RSVP_START_TIME`;
export const SET_RSVP_END_TIME = `${ PREFIX_TICKETS_STORE }/SET_RSVP_END_TIME`;
export const SET_RSVP_START_TIME_INPUT = `${ PREFIX_TICKETS_STORE }/SET_RSVP_START_TIME_INPUT`;
export const SET_RSVP_END_TIME_INPUT = `${ PREFIX_TICKETS_STORE }/SET_RSVP_END_TIME_INPUT`;

//
// ─── RSVP TEMP DETAILS TYPES ────────────────────────────────────────────────────
//

export const SET_RSVP_TEMP_TITLE = `${ PREFIX_TICKETS_STORE }/SET_RSVP_TEMP_TITLE`;
export const SET_RSVP_TEMP_DESCRIPTION = `${ PREFIX_TICKETS_STORE }/SET_RSVP_TEMP_DESCRIPTION`;
export const SET_RSVP_TEMP_CAPACITY = `${ PREFIX_TICKETS_STORE }/SET_RSVP_TEMP_CAPACITY`;
export const SET_RSVP_TEMP_NOT_GOING_RESPONSES = `${ PREFIX_TICKETS_STORE }/SET_RSVP_TEMP_NOT_GOING_RESPONSES`;
export const SET_RSVP_TEMP_START_DATE = `${ PREFIX_TICKETS_STORE }/SET_RSVP_TEMP_START_DATE`;
export const SET_RSVP_TEMP_START_DATE_INPUT = `${ PREFIX_TICKETS_STORE }/SET_RSVP_TEMP_START_DATE_INPUT`;
export const SET_RSVP_TEMP_START_DATE_MOMENT = `${ PREFIX_TICKETS_STORE }/SET_RSVP_TEMP_START_DATE_MOMENT`;
export const SET_RSVP_TEMP_END_DATE = `${ PREFIX_TICKETS_STORE }/SET_RSVP_TEMP_END_DATE`;
export const SET_RSVP_TEMP_END_DATE_INPUT = `${ PREFIX_TICKETS_STORE }/SET_RSVP_TEMP_END_DATE_INPUT`;
export const SET_RSVP_TEMP_END_DATE_MOMENT = `${ PREFIX_TICKETS_STORE }/SET_RSVP_TEMP_END_DATE_MOMENT`;
export const SET_RSVP_TEMP_START_TIME = `${ PREFIX_TICKETS_STORE }/SET_RSVP_TEMP_START_TIME`;
export const SET_RSVP_TEMP_END_TIME = `${ PREFIX_TICKETS_STORE }/SET_RSVP_TEMP_END_TIME`;
export const SET_RSVP_TEMP_START_TIME_INPUT = `${ PREFIX_TICKETS_STORE }/SET_RSVP_TEMP_START_TIME_INPUT`;
export const SET_RSVP_TEMP_END_TIME_INPUT = `${ PREFIX_TICKETS_STORE }/SET_RSVP_TEMP_END_TIME_INPUT`;

//
// ─── RSVP THUNK & SAGA TYPES ────────────────────────────────────────────────────
//

export const CREATE_RSVP = `${ PREFIX_TICKETS_STORE }/CREATE_RSVP`;
export const INITIALIZE_RSVP = `${ PREFIX_TICKETS_STORE }/INITIALIZE_RSVP`;
export const DELETE_RSVP = `${ PREFIX_TICKETS_STORE }/DELETE_RSVP`;

export const HANDLE_RSVP_START_DATE = `${ PREFIX_TICKETS_STORE }/HANDLE_RSVP_START_DATE`;
export const HANDLE_RSVP_END_DATE = `${ PREFIX_TICKETS_STORE }/HANDLE_RSVP_END_DATE`;
export const HANDLE_RSVP_START_TIME = `${ PREFIX_TICKETS_STORE }/HANDLE_RSVP_START_TIME`;
export const HANDLE_RSVP_END_TIME = `${ PREFIX_TICKETS_STORE }/HANDLE_RSVP_END_TIME`;

export const FETCH_RSVP_HEADER_IMAGE = `${ PREFIX_TICKETS_STORE }/FETCH_RSVP_HEADER_IMAGE`;
export const UPDATE_RSVP_HEADER_IMAGE = `${ PREFIX_TICKETS_STORE }/UPDATE_RSVP_HEADER_IMAGE`;
export const DELETE_RSVP_HEADER_IMAGE = `${ PREFIX_TICKETS_STORE }/DELETE_RSVP_HEADER_IMAGE`;
