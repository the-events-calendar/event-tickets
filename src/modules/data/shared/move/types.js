/**
 * Internal dependencies
 */
import { PREFIX_TICKETS_STORE } from '../../utils';

//
// ─── MODAL DATA ─────────────────────────────────────────────────────────────────
//

export const SET_MODAL_DATA = `${ PREFIX_TICKETS_STORE }/SET_MODAL_DATA`;
export const RESET_MODAL_DATA = `${ PREFIX_TICKETS_STORE }/RESET_MODAL_DATA`;
export const SUBMIT_MODAL = `${ PREFIX_TICKETS_STORE }/SUBMIT_MODAL`;

//
// ─── MODAL UI STATE ─────────────────────────────────────────────────────────────
//

export const INITIALIZE_MODAL = `${ PREFIX_TICKETS_STORE }/INITIALIZE_MODAL`;
export const SHOW_MODAL = `${ PREFIX_TICKETS_STORE }/SHOW_MODAL`;
export const HIDE_MODAL = `${ PREFIX_TICKETS_STORE }/HIDE_MODAL`;

//
// ─── API ───────────────────────────────────────────────────────────────────────-
//

export const FETCH_POST_TYPES = `${ PREFIX_TICKETS_STORE }/FETCH_POST_TYPES`;
export const FETCH_POST_TYPES_SUCCESS = `${ PREFIX_TICKETS_STORE }/FETCH_POST_TYPES_SUCCESS`;
export const FETCH_POST_TYPES_ERROR = `${ PREFIX_TICKETS_STORE }/FETCH_POST_TYPES_ERROR`;

export const FETCH_POST_CHOICES = `${ PREFIX_TICKETS_STORE }/FETCH_POST_CHOICES`;
export const FETCH_POST_CHOICES_SUCCESS = `${ PREFIX_TICKETS_STORE }/FETCH_POST_CHOICES_SUCCESS`;
export const FETCH_POST_CHOICES_ERROR = `${ PREFIX_TICKETS_STORE }/FETCH_POST_CHOICES_ERROR`;

export const MOVE_TICKET = `${ PREFIX_TICKETS_STORE }/MOVE_TICKET`;
export const MOVE_TICKET_SUCCESS = `${ PREFIX_TICKETS_STORE }/MOVE_TICKET_SUCCESS`;
export const MOVE_TICKET_ERROR = `${ PREFIX_TICKETS_STORE }/MOVE_TICKET_ERROR`;
