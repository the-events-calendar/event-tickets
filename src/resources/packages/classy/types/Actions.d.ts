import { Ticket } from './Ticket';

export type Actions = {
	// Tickets CRUD actions
	fetchTickets: (postId: number) => Promise<void>;
	createTicket: (ticketData: Partial<Ticket>) => Promise<void>;
	updateTicket: (ticketId: number, ticketData: Partial<Ticket>) => Promise<void>;
	deleteTicket: (ticketId: number) => Promise<void>;
	setCurrentPostId: (postId: number) => void;
	clearError: () => void;
};
