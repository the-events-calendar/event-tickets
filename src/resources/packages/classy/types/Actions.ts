import { Action } from 'redux';
import { TicketSettings } from './Ticket';

export const CREATE_TICKET = 'CREATE_TICKET';
export const DELETE_TICKET = 'DELETE_TICKET';
export const SET_IS_LOADING = 'SET_IS_LOADING';
export const SET_TICKETS = 'SET_TICKETS';
export const UPDATE_TICKET = 'UPDATE_TICKET';

export type CreateTicketAction = {
	ticket: TicketSettings;
} & Action<typeof CREATE_TICKET>;

export type DeleteTicketAction = {
	ticketId: number;
} & Action<typeof DELETE_TICKET>;

export type SetIsLoadingAction = {
	isLoading: boolean;
} & Action<typeof SET_IS_LOADING>;

export type SetTicketsAction = {
	tickets: TicketSettings[];
} & Action<typeof SET_TICKETS>;

export type UpdateTicketAction = {
	ticketId: number;
	ticketData: TicketSettings;
} & Action<typeof UPDATE_TICKET>;
