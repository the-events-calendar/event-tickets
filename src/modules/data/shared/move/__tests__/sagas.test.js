/**
 * External Dependencies
 */
import { call } from 'redux-saga/effects';

/**
 * Internal dependencies
 */
import * as types from '../types';
import watchers, * as sagas from '../sagas';

jest.mock( '@wordpress/data', () => ( {
	select() {
		return { getCurrentPostId: () => 1 };
	},
} ) );

describe( 'Move Sagas', () => {
	test( 'createBody', () => {
		expect( sagas.createBody( { a: 1, b: 2, c: 'okay&something' } ) ).toMatchSnapshot();
	} );

	describe( '_fetch', () => {
		it( 'should fetch successfully', () => {
			const gen = sagas._fetch( { a: 1 } );

			expect( gen.next().value ).toEqual( call(
				sagas.createBody, {
					a: 1,
					check: undefined,
				} ),
			);

			expect( gen.next( 'a=1&check=undefined' ).value ).toMatchSnapshot();
			expect( gen.next( { json() {} } ).value ).toMatchSnapshot();
			expect( gen.next().done ).toBeTruthy();
		} );
		it( 'should error', () => {
			const gen = sagas._fetch( { a: 1 } );
			expect( gen.next().value ).toMatchSnapshot();
			expect( gen.throw( 'error' ).value ).toMatchSnapshot();
			expect( gen.next().done ).toBeTruthy();
		} );
	} );

	describe( 'fetchPostTypes', () => {
		it( 'should fetch successfully', () => {
			const gen = sagas.fetchPostTypes();
			expect( gen.next().value ).toMatchSnapshot();
			expect( gen.next().value ).toMatchSnapshot();
			expect( gen.next( { data: {} } ).value ).toMatchSnapshot();
			expect( gen.next().done ).toBeTruthy();
		} );
		it( 'should error', () => {
			const gen = sagas.fetchPostTypes();
			expect( gen.next().value ).toMatchSnapshot();
			expect( gen.throw( 'error' ).value ).toMatchSnapshot();
			expect( gen.next().done ).toBeTruthy();
		} );
	} );

	describe( 'fetchPostChoices', () => {
		const params = {
			ignore: 1,
			post_type: 'all',
			search_terms: 'cool',
		};
		it( 'should fetch successfully', () => {
			const gen = sagas.fetchPostChoices( params );
			expect( gen.next().value ).toMatchSnapshot();
			expect( gen.next().value ).toMatchSnapshot();
			expect( gen.next( { data: {} } ).value ).toMatchSnapshot();
			expect( gen.next().done ).toBeTruthy();
		} );
		it( 'should error', () => {
			const gen = sagas.fetchPostChoices( params );
			expect( gen.next().value ).toMatchSnapshot();
			expect( gen.throw( 'error' ).value ).toMatchSnapshot();
			expect( gen.next().done ).toBeTruthy();
		} );
	} );

	describe( 'moveTicket', () => {
		const params = {
			src_post_id: 1,
			ticket_type_id: 2,
			target_post_id: 3,
		};
		it( 'should fetch successfully', () => {
			const gen = sagas.moveTicket( params );
			expect( gen.next().value ).toMatchSnapshot();
			expect( gen.next().value ).toMatchSnapshot();
			expect( gen.next( { data: {} } ).value ).toMatchSnapshot();
			expect( gen.next().done ).toBeTruthy();
		} );
		it( 'should error', () => {
			const gen = sagas.moveTicket( params );
			expect( gen.next().value ).toMatchSnapshot();
			expect( gen.throw( 'error' ).value ).toMatchSnapshot();
			expect( gen.next().done ).toBeTruthy();
		} );
	} );

	test( 'getCurrentPostId', () => {
		const gen = sagas.getCurrentPostId();
		expect( gen.next().value ).toMatchSnapshot();
	} );

	test( 'getPostChoices', () => {
		const gen = sagas.getPostChoices();
		expect( gen.next().value ).toMatchSnapshot();
		expect( gen.next( { post_type: 1, search_terms: '', ignore: 2 } ).value ).toMatchSnapshot();
		expect( gen.next().done ).toBeTruthy();
	} );

	describe( 'onModalChange', () => {
		let action;
		beforeEach( () => {
			action = {
				payload: {

				},
			};
		} );
		it( 'should get post choices', () => {
			const gen = sagas.onModalChange( action );
			expect( gen.next().value ).toMatchSnapshot();
			expect( gen.next().value ).toMatchSnapshot();
			expect( gen.next().done ).toBeTruthy();
		} );
		it( 'should not get post choices', () => {
			action.payload.ticketId = 1;
			const gen = sagas.onModalChange( action );
			expect( gen.next().done ).toBeTruthy();
		} );
	} );

	describe( 'onModalSubmit', () => {
		it( 'should hide modal on success', () => {
			const gen = sagas.onModalSubmit();
			expect( gen.next().value ).toMatchSnapshot();
			expect( gen.next( {
				src_post_id: 1,
				ticket_type_id: 2,
				target_post_id: 3,
			} ).value ).toMatchSnapshot();
			expect( gen.next().value ).toMatchSnapshot();
			expect( gen.next( { type: types.MOVE_TICKET_SUCCESS } ).value ).toMatchSnapshot();
			expect( gen.next().done ).toBeTruthy();
		} );
		it( 'should not hide modal on error', () => {
			const gen = sagas.onModalSubmit();
			expect( gen.next().value ).toMatchSnapshot();
			expect( gen.next( {
				src_post_id: 1,
				ticket_type_id: 2,
				target_post_id: 3,
			} ).value ).toMatchSnapshot();
			expect( gen.next().value ).toMatchSnapshot();
			expect( gen.next( { type: types.MOVE_TICKET_ERROR } ).value ).toMatchSnapshot();
			expect( gen.next().done ).toBeTruthy();
		} );
	} );

	test( 'onModalShow', () => {
		const gen = sagas.onModalShow( { payload: {} } );
		expect( gen.next().value ).toMatchSnapshot();
		expect( gen.next().done ).toBeTruthy();
	} );

	test( 'onModalHide', () => {
		const gen = sagas.onModalHide();
		expect( gen.next().value ).toMatchSnapshot();
		expect( gen.next().done ).toBeTruthy();
	} );

	test( 'initialize', () => {
		const gen = sagas.initialize();
		expect( gen.next().value ).toMatchSnapshot();
		expect( gen.next().done ).toBeTruthy();
	} );

	test( 'watchers', () => {
		const gen = watchers();
		expect( gen.next().value ).toMatchSnapshot();
		expect( gen.next().value ).toMatchSnapshot();
		expect( gen.next().value ).toMatchSnapshot();
		expect( gen.next().value ).toMatchSnapshot();
		expect( gen.next().value ).toMatchSnapshot();
		expect( gen.next().done ).toBeTruthy();
	} );
} );
