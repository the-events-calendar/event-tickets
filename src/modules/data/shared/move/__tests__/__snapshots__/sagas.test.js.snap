// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Move Sagas _fetch should error 1`] = `
{
  "@@redux-saga/IO": true,
  "CALL": {
    "args": [
      {
        "a": 1,
        "check": undefined,
      },
    ],
    "context": null,
    "fn": [Function],
  },
}
`;

exports[`Move Sagas _fetch should error 2`] = `undefined`;

exports[`Move Sagas _fetch should fetch successfully 1`] = `
{
  "@@redux-saga/IO": true,
  "CALL": {
    "args": [
      undefined,
      {
        "body": "a=1&check=undefined",
        "credentials": "include",
        "headers": {
          "Content-Type": "application/x-www-form-urlencoded;charset=UTF-8",
        },
        "method": "POST",
      },
    ],
    "context": null,
    "fn": [Function],
  },
}
`;

exports[`Move Sagas _fetch should fetch successfully 2`] = `
{
  "@@redux-saga/IO": true,
  "CALL": {
    "args": [],
    "context": {
      "json": [Function],
    },
    "fn": [Function],
  },
}
`;

exports[`Move Sagas createBody 1`] = `"a=1&b=2&c=okay%26something"`;

exports[`Move Sagas fetchPostChoices should error 1`] = `
{
  "@@redux-saga/IO": true,
  "PUT": {
    "action": {
      "type": "@@MT/TICKETS/FETCH_POST_CHOICES",
    },
    "channel": null,
  },
}
`;

exports[`Move Sagas fetchPostChoices should error 2`] = `
{
  "@@redux-saga/IO": true,
  "PUT": {
    "action": {
      "error": "error",
      "type": "@@MT/TICKETS/FETCH_POST_CHOICES_ERROR",
    },
    "channel": null,
  },
}
`;

exports[`Move Sagas fetchPostChoices should fetch successfully 1`] = `
{
  "@@redux-saga/IO": true,
  "PUT": {
    "action": {
      "type": "@@MT/TICKETS/FETCH_POST_CHOICES",
    },
    "channel": null,
  },
}
`;

exports[`Move Sagas fetchPostChoices should fetch successfully 2`] = `
{
  "@@redux-saga/IO": true,
  "CALL": {
    "args": [
      {
        "action": "move_tickets_get_post_choices",
        "ignore": 1,
        "post_type": "all",
        "search_terms": "cool",
      },
    ],
    "context": null,
    "fn": [Function],
  },
}
`;

exports[`Move Sagas fetchPostChoices should fetch successfully 3`] = `
{
  "@@redux-saga/IO": true,
  "PUT": {
    "action": {
      "data": {},
      "type": "@@MT/TICKETS/FETCH_POST_CHOICES_SUCCESS",
    },
    "channel": null,
  },
}
`;

exports[`Move Sagas fetchPostTypes should error 1`] = `
{
  "@@redux-saga/IO": true,
  "PUT": {
    "action": {
      "type": "@@MT/TICKETS/FETCH_POST_TYPES",
    },
    "channel": null,
  },
}
`;

exports[`Move Sagas fetchPostTypes should error 2`] = `
{
  "@@redux-saga/IO": true,
  "PUT": {
    "action": {
      "error": "error",
      "type": "@@MT/TICKETS/FETCH_POST_TYPES_ERROR",
    },
    "channel": null,
  },
}
`;

exports[`Move Sagas fetchPostTypes should fetch successfully 1`] = `
{
  "@@redux-saga/IO": true,
  "PUT": {
    "action": {
      "type": "@@MT/TICKETS/FETCH_POST_TYPES",
    },
    "channel": null,
  },
}
`;

exports[`Move Sagas fetchPostTypes should fetch successfully 2`] = `
{
  "@@redux-saga/IO": true,
  "CALL": {
    "args": [
      {
        "action": "move_tickets_get_post_types",
      },
    ],
    "context": null,
    "fn": [Function],
  },
}
`;

exports[`Move Sagas fetchPostTypes should fetch successfully 3`] = `
{
  "@@redux-saga/IO": true,
  "PUT": {
    "action": {
      "data": {},
      "type": "@@MT/TICKETS/FETCH_POST_TYPES_SUCCESS",
    },
    "channel": null,
  },
}
`;

exports[`Move Sagas getCurrentPostId 1`] = `
{
  "@@redux-saga/IO": true,
  "CALL": {
    "args": [],
    "context": {
      "getCurrentPostId": [Function],
    },
    "fn": [Function],
  },
}
`;

exports[`Move Sagas getPostChoices 1`] = `
{
  "@@redux-saga/IO": true,
  "ALL": {
    "ignore": {
      "@@redux-saga/IO": true,
      "CALL": {
        "args": [],
        "context": null,
        "fn": [Function],
      },
    },
    "post_type": {
      "@@redux-saga/IO": true,
      "SELECT": {
        "args": [],
        "selector": [Function],
      },
    },
    "search_terms": {
      "@@redux-saga/IO": true,
      "SELECT": {
        "args": [],
        "selector": [Function],
      },
    },
  },
}
`;

exports[`Move Sagas getPostChoices 2`] = `
{
  "@@redux-saga/IO": true,
  "CALL": {
    "args": [
      {
        "ignore": 2,
        "post_type": 1,
        "search_terms": "",
      },
    ],
    "context": null,
    "fn": [Function],
  },
}
`;

exports[`Move Sagas initialize 1`] = `
{
  "@@redux-saga/IO": true,
  "ALL": [
    {
      "@@redux-saga/IO": true,
      "CALL": {
        "args": [],
        "context": null,
        "fn": [Function],
      },
    },
    {
      "@@redux-saga/IO": true,
      "CALL": {
        "args": [],
        "context": null,
        "fn": [Function],
      },
    },
  ],
}
`;

exports[`Move Sagas moveTicket should error 1`] = `
{
  "@@redux-saga/IO": true,
  "PUT": {
    "action": {
      "type": "@@MT/TICKETS/MOVE_TICKET",
    },
    "channel": null,
  },
}
`;

exports[`Move Sagas moveTicket should error 2`] = `
{
  "@@redux-saga/IO": true,
  "PUT": {
    "action": {
      "error": "error",
      "type": "@@MT/TICKETS/MOVE_TICKET_ERROR",
    },
    "channel": null,
  },
}
`;

exports[`Move Sagas moveTicket should fetch successfully 1`] = `
{
  "@@redux-saga/IO": true,
  "PUT": {
    "action": {
      "type": "@@MT/TICKETS/MOVE_TICKET",
    },
    "channel": null,
  },
}
`;

exports[`Move Sagas moveTicket should fetch successfully 2`] = `
{
  "@@redux-saga/IO": true,
  "CALL": {
    "args": [
      {
        "action": "move_ticket_type",
        "src_post_id": 1,
        "target_post_id": 3,
        "ticket_type_id": 2,
      },
    ],
    "context": null,
    "fn": [Function],
  },
}
`;

exports[`Move Sagas moveTicket should fetch successfully 3`] = `
{
  "@@redux-saga/IO": true,
  "PUT": {
    "action": {
      "data": {},
      "type": "@@MT/TICKETS/MOVE_TICKET_SUCCESS",
    },
    "channel": null,
  },
}
`;

exports[`Move Sagas onModalChange should get post choices 1`] = `
{
  "@@redux-saga/IO": true,
  "CALL": {
    "args": [
      500,
    ],
    "context": null,
    "fn": [Function],
  },
}
`;

exports[`Move Sagas onModalChange should get post choices 2`] = `
{
  "@@redux-saga/IO": true,
  "CALL": {
    "args": [],
    "context": null,
    "fn": [Function],
  },
}
`;

exports[`Move Sagas onModalHide 1`] = `
{
  "@@redux-saga/IO": true,
  "PUT": {
    "action": {
      "type": "@@MT/TICKETS/RESET_MODAL_DATA",
    },
    "channel": null,
  },
}
`;

exports[`Move Sagas onModalShow 1`] = `
{
  "@@redux-saga/IO": true,
  "PUT": {
    "action": {
      "payload": {},
      "type": "@@MT/TICKETS/SET_MODAL_DATA",
    },
    "channel": null,
  },
}
`;

exports[`Move Sagas onModalSubmit should hide modal on success 1`] = `
{
  "@@redux-saga/IO": true,
  "ALL": {
    "src_post_id": {
      "@@redux-saga/IO": true,
      "CALL": {
        "args": [],
        "context": null,
        "fn": [Function],
      },
    },
    "target_post_id": {
      "@@redux-saga/IO": true,
      "SELECT": {
        "args": [],
        "selector": [Function],
      },
    },
    "ticket_type_id": {
      "@@redux-saga/IO": true,
      "SELECT": {
        "args": [],
        "selector": [Function],
      },
    },
  },
}
`;

exports[`Move Sagas onModalSubmit should hide modal on success 2`] = `
{
  "@@redux-saga/IO": true,
  "FORK": {
    "args": [
      {
        "src_post_id": 1,
        "target_post_id": 3,
        "ticket_type_id": 2,
      },
    ],
    "context": null,
    "fn": [Function],
  },
}
`;

exports[`Move Sagas onModalSubmit should hide modal on success 3`] = `
{
  "@@redux-saga/IO": true,
  "TAKE": {
    "pattern": [
      "@@MT/TICKETS/MOVE_TICKET_SUCCESS",
      "@@MT/TICKETS/MOVE_TICKET_ERROR",
    ],
  },
}
`;

exports[`Move Sagas onModalSubmit should hide modal on success 4`] = `
{
  "@@redux-saga/IO": true,
  "PUT": {
    "action": {
      "type": "@@MT/TICKETS/HIDE_MODAL",
    },
    "channel": null,
  },
}
`;

exports[`Move Sagas onModalSubmit should not hide modal on error 1`] = `
{
  "@@redux-saga/IO": true,
  "ALL": {
    "src_post_id": {
      "@@redux-saga/IO": true,
      "CALL": {
        "args": [],
        "context": null,
        "fn": [Function],
      },
    },
    "target_post_id": {
      "@@redux-saga/IO": true,
      "SELECT": {
        "args": [],
        "selector": [Function],
      },
    },
    "ticket_type_id": {
      "@@redux-saga/IO": true,
      "SELECT": {
        "args": [],
        "selector": [Function],
      },
    },
  },
}
`;

exports[`Move Sagas onModalSubmit should not hide modal on error 2`] = `
{
  "@@redux-saga/IO": true,
  "FORK": {
    "args": [
      {
        "src_post_id": 1,
        "target_post_id": 3,
        "ticket_type_id": 2,
      },
    ],
    "context": null,
    "fn": [Function],
  },
}
`;

exports[`Move Sagas onModalSubmit should not hide modal on error 3`] = `
{
  "@@redux-saga/IO": true,
  "TAKE": {
    "pattern": [
      "@@MT/TICKETS/MOVE_TICKET_SUCCESS",
      "@@MT/TICKETS/MOVE_TICKET_ERROR",
    ],
  },
}
`;

exports[`Move Sagas onModalSubmit should not hide modal on error 4`] = `undefined`;

exports[`Move Sagas watchers 1`] = `
{
  "@@redux-saga/IO": true,
  "FORK": {
    "args": [
      [
        "@@MT/TICKETS/INITIALIZE_MODAL",
      ],
      [Function],
    ],
    "context": null,
    "fn": [Function],
  },
}
`;

exports[`Move Sagas watchers 2`] = `
{
  "@@redux-saga/IO": true,
  "FORK": {
    "args": [
      [
        "@@MT/TICKETS/SET_MODAL_DATA",
      ],
      [Function],
    ],
    "context": null,
    "fn": [Function],
  },
}
`;

exports[`Move Sagas watchers 3`] = `
{
  "@@redux-saga/IO": true,
  "FORK": {
    "args": [
      [
        "@@MT/TICKETS/SUBMIT_MODAL",
      ],
      [Function],
    ],
    "context": null,
    "fn": [Function],
  },
}
`;

exports[`Move Sagas watchers 4`] = `
{
  "@@redux-saga/IO": true,
  "FORK": {
    "args": [
      [
        "@@MT/TICKETS/SHOW_MODAL",
      ],
      [Function],
    ],
    "context": null,
    "fn": [Function],
  },
}
`;

exports[`Move Sagas watchers 5`] = `
{
  "@@redux-saga/IO": true,
  "FORK": {
    "args": [
      [
        "@@MT/TICKETS/HIDE_MODAL",
      ],
      [Function],
    ],
    "context": null,
    "fn": [Function],
  },
}
`;
